panther_manager/test/test_lights_behavior_tree.cpp

```diff
@@ -0,0 +1,268 @@
+    SetLEDAnimationSrv::Response::SharedPtr response);
+
+  rclcpp::Publisher<BoolMsg>::SharedPtr e_stop_pub_;
+  rclcpp::Publisher<BatteryStateMsg>::SharedPtr battery_state_pub_;
+  rclcpp::Service<SetLEDAnimationSrv>::SharedPtr set_led_animation_server_;
+};
+
+TestLightsBehaviorTree::TestLightsBehaviorTree()
+{
+  using namespace std::placeholders;
+
+  std::vector<std::string> plugin_libs;
+  plugin_libs.push_back("tick_after_timeout_bt_node");
+
+  std::vector<std::string> ros_plugin_libs;
+  ros_plugin_libs.push_back("call_set_led_animation_service_bt_node");
+
+  std::vector<rclcpp::Parameter> params;
+  params.push_back(rclcpp::Parameter("plugin_libs", plugin_libs));
+  params.push_back(rclcpp::Parameter("ros_plugin_libs", ros_plugin_libs));
+  params.push_back(rclcpp::Parameter("low_battery_anim_period", kLowBatteryAnimPeriod));
+  params.push_back(rclcpp::Parameter("critical_battery_anim_period", kCriticalBatteryAnimPeriod));
+
+  rclcpp::NodeOptions options;
+  options.parameter_overrides(params);
+
+  lights_manager_node_ = std::make_shared<LightsManagerNodeWrapper>(
+    "test_lights_manager_node", options);
+
+  e_stop_pub_ = lights_manager_node_->create_publisher<BoolMsg>(
+    "hardware/e_stop", rclcpp::QoS(rclcpp::KeepLast(1)).transient_local().reliable());
+  battery_state_pub_ = lights_manager_node_->create_publisher<BatteryStateMsg>("battery", 3);
+  set_led_animation_server_ = lights_manager_node_->create_service<SetLEDAnimationSrv>(
+    "lights/controller/set/animation",
+    std::bind(&TestLightsBehaviorTree::SetLEDAnimationCB, this, _1, _2));
+}
+
+TestLightsBehaviorTree::~TestLightsBehaviorTree() {}
+
+void TestLightsBehaviorTree::SpinWhileRunning()
+{
+  test_behavior_tree_utils::SpinWhileRunning(
+    lights_manager_node_, [&]() { return lights_manager_node_->GetLightsTreeStatus(); });
+}
+
+void TestLightsBehaviorTree::PublishEStop(const bool data)
+{
+  BoolMsg msg;
+  msg.data = data;
+  e_stop_pub_->publish(msg);
+}
+
+void TestLightsBehaviorTree::PublishBatteryState(
+  const std::uint8_t status, const std::uint8_t health, const float percentage)
+{
+  BatteryStateMsg msg;
+  msg.power_supply_status = status;
+  msg.power_supply_health = health;
+  msg.percentage = percentage;
+  battery_state_pub_->publish(msg);
+}
+
+void TestLightsBehaviorTree::SetLEDAnimationCB(
+  const SetLEDAnimationSrv::Request::SharedPtr & request,
+  SetLEDAnimationSrv::Response::SharedPtr response)
+{
+  current_anim_id_ = request->animation.id;
+  response->success = true;
+}
+
+TEST_F(TestLightsBehaviorTree, UnknownBatteryStatus)
+{
+  lights_manager_node_->Initialize();
+  PublishEStop(false);
+  PublishBatteryState(
+    BatteryStateMsg::POWER_SUPPLY_STATUS_UNKNOWN, BatteryStateMsg::POWER_SUPPLY_HEALTH_GOOD, 1.0);
+
+  SpinWhileRunning();
+  EXPECT_EQ(current_anim_id_, LEDAnimationMsg::ERROR);
+}
+
+TEST_F(TestLightsBehaviorTree, NotCharging)
+{
+  lights_manager_node_->Initialize();
```

Comment:
Initialization is repeated on each test case. How about adding this to the constructor of the test fixture?